Feature/rename chef solo

dereklwood opened the pull request:
Renaming the Chef Automator plugin to the Chef Solo Automator plugin, and corresponding automatortype from "chef" to "chef-solo".  

This updates:
- [x] provisioner plugin code
- [x] config defaults and examples
- [x] documentation
- [x] unit tests for UI and Server

Commit:
rename chef_automator to chef_solo_automator

Commit:
rename config defaults automatortype to chef-solo

Commit:
doc updates for renaming chef automator to chef-solo automator

Commit:
updating ui tests for renaming chef to chef-solo

Commit:
Merge branch 'develop' into feature/rename-chef-solo

Commit:
fix local primitives path

Commit:
updating java unit tests for chef solo rename

wolf31o2 commented on config/defaults/services/nodejs.json:
```diff
@@ -6,7 +6,7 @@
     "actions": {
       "install": {
-        "type":"chef",
+        "type":"chef-solo-solo",
```
There can be only one!

wolf31o2 commented on docs/docs/source/guide/admin/chef-solo-automator-plugin.rst:
```diff
@@ -154,39 +154,39 @@ Each Loom Automator plugin is responsible for implementing a bootstrap method in
-Since the Chef Automator plugin is implemented using chef-solo, the following restrictions apply:
+Since the Chef Solo Automator plugin is implemented using chef-solo, the following restrictions apply:
 
 	* No chef search capability
```
Not a part of the diff, exactly, but let's capitalize Chef

wolf31o2 commented on docs/docs/source/guide/admin/chef-solo-automator-plugin.rst:
```diff
@@ -207,7 +207,7 @@ all the nodes in the cluster. It then simply utilizes the community "hostsfile"
 each node.
 
 The example loom service definition invoking this cookbook is called "hosts". It simply sets up a "configure" service
```
This is done by base, now, not hosts.

wolf31o2 commented:
A couple comments... only 1 error that needs to be fixed. Otherwise, LGTM.

Commit:
fixing double substitution

Commit:
PR feedback, plus quick section on loom_base

dereklwood commented:
addressed comments... needed to add a quick section on recipe loom_base, which should be reviewed: https://github.com/continuuity/loom/commit/66ea4dac312f21de3d2422165a143ffbe8ad6696

wolf31o2 commented:
New loom_base looks good. :+1:
